### PR TITLE
feat: upgrade for @kaiu/serializer v1.0.1

### PR DESCRIPTION
**Pull request by Supamiu**
This PR contains a lot of things:

- `forChild` addition.
- Travis fixes & codecov fixes.
- @kaiu/serializer v1.0.1 compatibility.
- Renaming.

### CODE REVIEW

File: src/ng-serializer.factory.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
...
-import { Registration } from '@kaiu/serializer';
+import { Registration, Registry } from '@kaiu/serializer';
 
-export function NgSerializerServiceFactory(registrations?: Registration[]): NgSerializerService {
-    const service: NgSerializerService = new NgSerializerService();
-    service.register(registrations);
-    return service;
+export function NgSerializerServiceFactory(registrations?: Registration[][]): NgSerializerService {
+    const registry: Registry = new Registry();
+    //The concat called here is used to flatten the array., allowing multi:true in provider.
```
Comment (noemi-salaun) [inline src/ng-serializer.factory.ts]: There is a comma typo

File: src/ng-serializer.service.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import { Injectable } from '@angular/core';
-import { Serializer } from '@kaiu/serializer';
+import { Serializer, Registry } from '@kaiu/serializer';
 
 @Injectable()
 export class NgSerializerService extends Serializer {
+    constructor(public readonly registry: Registry) {
```
Comment (noemi-salaun) [inline src/ng-serializer.service.ts]: No need to make the registry public readonly in this constructor, the Parent constructor will take care of it

**Review by noemi-salaun — CHANGES_REQUESTED** [review]

File: src/ng-serializer.service.ts (hunk 2 above)
Comment (Supamiu) [reply src/ng-serializer.service.ts]: This has been done because else aot can't resolve all parameters for this service.

Comment (noemi-salaun) [reply src/ng-serializer.service.ts]: You can inject the registry and then pass it to the parent constructor

```ts
constructor(registry: Registry) {
    super(registry);
}
```

**Review by noemi-salaun — CHANGES_REQUESTED** [review]

**Review by noemi-salaun — APPROVED** [review]

**Pull request merged**